### DISCUSSION TITLE
Dump record

### DISCUSSION THREAD

**Discussion by tao363** [FreeRDP/FreeRDP#10493] [Q&A]
The current version is 3.3.0. What do I need to do to enable dump on the client and proxy?

**Comment by akallabeth** [comment]
you need to enable the dump settings `FreeRDP_TransportDump` and `FreeRDP_TransportDumpFile` for each connection established.
this is currently not implemented and you need to do that with a custom module or patches in source to generate these.
only with `xfreerdp` you can enable this by command line

**Reply by tao363** [reply]
I hardcoded FreeRDP TransportDump = True and FreeRDP TransportDumpFile ="D:\\data" in the wfreerdp.c file, but no file was saved after the startup

**Reply by akallabeth** [reply]
well, I doubt you started the client?
you should modify the proxy if you want to capture there.

**Reply by tao363** [reply]
I saw this discussion: [#8769](https://github.com/FreeRDP/FreeRDP/discussions/8769)
My goal is to try to implement this in server/proxy and understand the implementation details.
Please, what should I do next to achieve this?

**Reply by tao363** [reply]
I implemented the recording as discussed, but the stored file is still large. What is the cause of this?
My understanding is that using StreamDump should be some drawing instructions and some bitmaps. The amount of data stored should be small. This is confusing me. I look forward to your answer.

**Reply by Mr-LZG** [reply]
@tao363 Please tell me how to implement this function
